src/main/kotlin/org/kamiblue/client/gui/hudgui/elements/misc/TPS.kt

```diff
@@ -11,14 +12,20 @@ internal object TPS : LabelHud(
     description = "Server TPS"
 ) {
 
+    private val mspt = setting("Show MSPT", false)
+
     // buffered TPS readings to add some fluidity to the TPS HUD element
     private val tpsBuffer = CircularArray.create(20, 20f)
 
     override fun SafeClientEvent.updateText() {
         tpsBuffer.add(TpsCalculator.tickRate)
-
-        displayText.add("%.2f".format(tpsBuffer.average()), primaryColor)
-        displayText.add("tps", secondaryColor)
+        if (mspt.value) {
+            displayText.add("%.2f".format(1000 / tpsBuffer.average()), primaryColor)
```

Comment:
While this would get you the average, it won't give you the raw in-between values from the TpsCalculator. You may want to look at migrating the tpsBuffer to be a millisecond counter buffer instead with 50 as the default. The main issue is that MSPT.

My issue with this entire attempt is that MSPT when TPS is 20 is going to be inaccurate as the server side MSPT will be anywhere between 0 and 50 for 20TPS. The other variations are easily figure out as a slower tick rate is always playing catch up.

This method should displace "<= 50" or similar to indicate that we are not able to accurately display the MSPT in the HUD without having some source of TPS such as the tab list or a command being ran which would be slightly more active in the parsing.

Reply:
Very true, the issue is I'm not sure how to properly implement this, other than a messy check if MSPT <= 50.